pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/bestpractices/LiteralsFirstInComparisonsRule.java

```diff
@@ -162,17 +159,13 @@ private boolean isStringLiteral(JavaNode node) {
     private boolean isConstantString(JavaNode node) {
         if (node instanceof ASTName) {
             ASTName name = (ASTName) node;
-            ASTClassOrInterfaceBody classBody = name.getFirstParentOfType(ASTClassOrInterfaceBody.class);
-            ASTClassOrInterfaceBodyDeclaration classOrInterfaceBodyDeclaration = classBody.getFirstChildOfType(ASTClassOrInterfaceBodyDeclaration.class);
```

Comment:
The problem was that here, we only looked for the first declaration. So only the first field declaration was considered.